### PR TITLE
Add USE_PR_BODY to controll whether to use PR bodies for feat and fix

### PR DESCRIPTION
**Pull request by spetrosi**

### CODE REVIEW

**Comment by spetrosi** [comment]
I was not very careful with trailing newlines because markdown does not care about multiple blank lines. But seems like people tend to view raw file too.

File: role-make-version-changelog.sh [hunk 1]
```diff
@@ -296,6 +297,8 @@ You have three options:
         # When PR are edited using GH web UI, DOS carriage return ^M appears
         # in PR title and body
         sed -i "s/$(printf '\r')\$//" "$rel_notes_file"
+        # $(< file) strips trailing blank lines from a file
+        a=$(<"$rel_notes_file"); printf '%s\n' "$a" > "$rel_notes_file"
```
Comment (richm) [inline role-make-version-changelog.sh:301]: is there somewhere else that adds the single blank line between the new entry and the old entry?

File: role-make-version-changelog.sh [hunk 2]
```diff
@@ -152,7 +152,8 @@ else
         pr_num="$(echo "$pr_entry" | jq -r '.number')"
         pr_body="$(echo "$pr_entry" | jq -r '.body')"
         echo "$pr_title" >> "$pr_titles_file"
-        if [ -n "$pr_body" ]; then
+        # Only append PR body to PRs of type feat and fix
+        if [ -n "$pr_body" ] && echo "$pr_title" | sed -n 1p | grep -e '^- feat.*' -e '^- fix.*'; then
```
Comment (richm) [inline role-make-version-changelog.sh:156]:
```suggestion
        if [ -n "$pr_body" ] && [[ "$pr_title" =~ ^(feat|fix) ]]; then
```
will a pr_title be multi-line?

**Review by richm — DISMISSED** [review]

File: role-make-version-changelog.sh [hunk 3]
```diff
@@ -152,12 +152,16 @@ else
...
         pr_body="$(echo "$pr_entry" | jq -r '.body')"
         echo "$pr_title" >> "$pr_titles_file"
-        if [ -n "$pr_body" ]; then
+        # Only append PR body to PRs of type feat and fix
+        if [ -n "$pr_body" ] && echo "$pr_title" | grep -e '^feat.*' -e '^fix.*'; then
             # Indent pr_body 2 spaces to act as a bullet content for pr_title
             pr_body_ind="$(echo "$pr_body"  | awk '{ print "  " $0 }')"
-            printf -v pr_description -- "- %s (#%s)\n\n%s\n" "$pr_title" "$pr_num" "$pr_body_ind"
+            # Remove DOS carriage return ^M from PR body
+            # DOS carriage return ^M appears when PRs are edited using GH web UI
```
Comment (richm) [inline role-make-version-changelog.sh:160]:
```suggestion
            # DOS carriage return ^M appears when PRs are edited using GH web UI
            # shellcheck disable=SC2001
```

**Review by richm — APPROVED** [review]

**Pull request merged**